File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    let right_arr = []
    let real_arr = arr.length
    for (let i = 0 ; i < real_arr; i++){
        if(arr[i]!==arr[i+1]){
            right_arr.push(arr[i])
        }
    }
    return right_arr;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque
def solution(arr):
    left = []
    temp = arr.copy()
    right = deque(temp)
    
    for _ in arr:
        test = right.popleft()
        if not left:
            left.append(test)
            continue
        if left[-1] != test:
            left.append(test)
        else:
            pass
    return left